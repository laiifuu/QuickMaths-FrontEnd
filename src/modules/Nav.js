import { useState, useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { NavLink, useNavigate } from 'react-router-dom';
import { logout } from '../redux/user/session-redux';

const Nav = () => {
  const dispatch = useDispatch();

  const navigate = useNavigate();

  const [showMenu, setShowMenu] = useState(false);

  const isLoggedIn = JSON.parse(window.localStorage.getItem('logged_in'));

  useEffect(() => {
    window.addEventListener('resize', () => {
      if (window.innerWidth >= 768) {
        setShowMenu(true);
      } else {
        setShowMenu(false);
      }
      const body = document.querySelector('body');
      body.classList.remove('no-scroll');
    });
  }, []);

  const toggleMenu = () => {
    if (window.innerWidth < 768) {
      setShowMenu(!showMenu);
      const body = document.querySelector('body');
      body.classList.toggle('no-scroll');
    }
  };

  const keyInput = (e) => {
    if (e.code === 'Enter') {
      toggleMenu();
    }
  };

  const clearUserData = () => {
    localStorage.clear();
    dispatch(logout());
    toggleMenu();
    navigate('/');
  };

  return (
    <header>
      <i
        className={
          !showMenu
            ? 'menuButton fa-solid fa-bars '
            : 'menuButton fa-solid fa-xmark'
        }
        onClick={toggleMenu}
        onKeyDown={keyInput}
        tabIndex="0"
        aria-label="burger menu button"
        role="button"
      />

      <nav className={!showMenu ? 'display-none' : ''}>
        <NavLink to="/" className="logo" onClick={toggleMenu}>
          QuickMaths
        </NavLink>

        {isLoggedIn ? (
          <ul className="menuLinks">
            <li>
              <NavLink to="/" onClick={toggleMenu}>
                Tutors
              </NavLink>
            </li>
            <li>
              <NavLink to="/reserve" onClick={toggleMenu}>
                Reserve
              </NavLink>
            </li>
            <li>
              <NavLink to="/reservations" onClick={toggleMenu}>
                My reservations
              </NavLink>
            </li>
            <li>
              <NavLink to="/add_tutor" onClick={toggleMenu}>
                Add tutor
              </NavLink>
            </li>
            <li>
              <NavLink to="/delete_tutor" onClick={toggleMenu}>
                Delete tutor
              </NavLink>
            </li>
          </ul>
        ) : (

          <NavLink
            to="/user/login"
            onClick={toggleMenu}
          >
            <button className="login-button" type="button">
              Log in
            </button>
          </NavLink>
        )}

        <div className="logout-sm">
          {isLoggedIn && (
            <button
              className="logout-btn"
              type="button"
              onClick={clearUserData}
            >
              Logout
            </button>
          )}

          <ul className="socialLinks">
            <li>
              <a
                href="https://facebook.com"
                onClick={toggleMenu}
                target="_blank"
                rel="noreferrer"
                aria-label="facebook page"
              >
                <i className="fa-brands fa-facebook-f" />
              </a>
            </li>
            <li>
              <a
                href="https://twitter.com"
                onClick={toggleMenu}
                target="_blank"
                rel="noreferrer"
                aria-label="twitter page"
              >
                <i className="fa-brands fa-twitter" />
              </a>
            </li>
            <li>
              <a
                href="https://instagram.com"
                onClick={toggleMenu}
                target="_blank"
                rel="noreferrer"
                aria-label="instagram page"
              >
                <i className="fa-brands fa-instagram" />
              </a>
            </li>
          </ul>
        </div>

        <div className="navRights">All rights reserved</div>
      </nav>
    </header>
  );
};

export default Nav;
