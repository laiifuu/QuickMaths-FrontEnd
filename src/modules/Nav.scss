* {
  font-weight: 900;
  color: rgb(51, 51, 51);
}

nav {
  font-size: 1.5rem;
  background-color: white;
  position: absolute;
  display: flex;
  flex-direction: column;
  margin: 0;
  height: 100vh;
  width: 100vw;
}

.display-none {
  display: none;
}

.logo {
  text-align: center;
  padding: 1rem;
}

.menuButton {
  position: absolute;
  left: 1rem;
  top: 1rem;
}

.menuLinks {
  margin-top: 2rem;

  li {
    padding: 0.5rem;
    text-align: center;
  }
}

.socialLinks {
  margin-top: auto;
  display: flex;
  justify-content: center;

  li {
    padding: 1rem;
  }
}

.navRights {
  text-align: center;
  font-size: 0.8rem;
  padding-bottom: 1rem;
}

.navAngleClose {
  display: none;
}

@media only screen and (min-width: 768px) {
  nav {
    font-size: 1rem;
    float: left;
    display: flex;
    flex-direction: column;
    margin: 0;
    height: 100vh;
    width: 10rem;
    box-shadow: 1px 1px 2px lightgray;
    background-color: white;
    position: static;
  }

  .navAngleClose {
    display: block;
    top: 80%;
    left: 10rem;
    color: white;
    background-color: lightgreen;
  }

  .closeMenuButton {
    display: none;
  }

  .navAngle {
    position: absolute;
    padding: 0.7rem 0.7rem 0.7rem 1.5rem;
    border-radius: 0 1rem 1rem 0;
  }

  .menuLinks {
    li {
      text-align: left;
      margin-left: 1rem;
    }

    li:hover {
      background-color: lightgreen;

      a {
        color: white;
      }
    }
  }
}
