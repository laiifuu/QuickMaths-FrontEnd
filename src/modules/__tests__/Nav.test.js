import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/configureStore';
import Nav from '../Nav';

describe('Nav', () => {
  test('Snap the Nav', async () => {
    const snapNav = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(snapNav).toMatchSnapshot();
  });
});
