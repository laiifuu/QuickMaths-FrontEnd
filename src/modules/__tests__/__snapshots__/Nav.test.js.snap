// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Snap the Nav 1`] = `
<header>
  <i
    aria-label="burger menu button"
    className="menuButton fa-solid fa-bars "
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex="0"
  />
  <nav
    className="display-none"
  >
    <a
      aria-current="page"
      className="logo active"
      href="/"
      onClick={[Function]}
    >
      QuickMaths
    </a>
    <a
      className=""
      href="/user/login"
      onClick={[Function]}
    >
      <button
        className="login-button"
        type="button"
      >
        Log in
      </button>
    </a>
    <div
      className="logout-sm"
    >
      <ul
        className="socialLinks"
      >
        <li>
          <a
            aria-label="facebook page"
            href="https://facebook.com"
            onClick={[Function]}
            rel="noreferrer"
            target="_blank"
          >
            <i
              className="fa-brands fa-facebook-f"
            />
          </a>
        </li>
        <li>
          <a
            aria-label="twitter page"
            href="https://twitter.com"
            onClick={[Function]}
            rel="noreferrer"
            target="_blank"
          >
            <i
              className="fa-brands fa-twitter"
            />
          </a>
        </li>
        <li>
          <a
            aria-label="instagram page"
            href="https://instagram.com"
            onClick={[Function]}
            rel="noreferrer"
            target="_blank"
          >
            <i
              className="fa-brands fa-instagram"
            />
          </a>
        </li>
      </ul>
    </div>
    <div
      className="navRights"
    >
      All rights reserved
    </div>
  </nav>
</header>
`;
